### PR TITLE
MAINT: Fixed several unused imports and unused assignments in scipy/interpolate

### PR DESCRIPTION
**Pull request by pvanmulbregt**
#### Reference issue
Further work on gh-11529

#### What does this implement/fix?
To reduce Pyflakes warnings, remove five(5) unused imports; eleven (11) unused variable assignments; commented on why one apparently unused import needs to be kept;

### CODE REVIEW

File: scipy/interpolate/tests/test_polyint.py [hunk 1]
```diff
@@ -198,14 +198,14 @@ def test_high_derivative(self):
...
-              self.true_poly(2)]
+        # xs = [0,0,0,1,1,1,2]
+        # ys = [self.true_poly(0),
+        #       self.true_poly.deriv(1)(0),
+        #       self.true_poly.deriv(2)(0),
+        #       self.true_poly(1),
+        #       self.true_poly.deriv(1)(1),
+        #       self.true_poly.deriv(2)(1),
+        #       self.true_poly(2)]
         P = KroghInterpolator(self.xs,self.ys)
```
Comment (pvanmulbregt) [inline scipy/interpolate/tests/test_polyint.py]: @aarchiba What is the purpose of the `xs`, `ys` variables in this test? They appear to be unused  with `self.xs, self.ys` used instead.  Should the test use `xs`?  Or should  `xs`, `ys` be just completely removed? Thanks!

File: scipy/interpolate/fitpack.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 import numpy as np
 
+# These are in the API for fftpack even if not used in fftpacvk.py itself.
```
Comment (ev-br) [inline scipy/interpolate/fitpack.py]: Minor: fitpack, not fftpack

File: scipy/interpolate/tests/test_fitpack.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def check_1(self,f=f1,per=0,s=0,a=0,b=2*pi,N=20,at=0,xb=None,xe=None):
             xe = b
         x = a+(b-a)*arange(N+1,dtype=float)/float(N)    # nodes
         x1 = a+(b-a)*arange(1,N,dtype=float)/float(N-1)  # middle points of the nodes
-        v,v1 = f(x),f(x1)
+        v, _ = f(x),f(x1)
```
Comment (ev-br) [inline scipy/interpolate/tests/test_fitpack.py]: If f(x1) is not used, maybe just avoid computing it

File: scipy/interpolate/tests/test_interpolate.py [hunk 4]
```diff
@@ -913,7 +913,7 @@ def test_axis(self):
         c_s = c.shape
         xp = np.random.random((1, 2))
         for axis in (0, 1, 2, 3):
-            k, m = c.shape[axis], c.shape[axis+1]
+            _, m = c.shape[axis], c.shape[axis+1]
```
Comment (ev-br) [inline scipy/interpolate/tests/test_interpolate.py]: Ditto, if c.shape[axis] is not used, then drop it.

File: scipy/interpolate/tests/test_polyint.py (hunk 1 above)
Comment (ev-br) [reply scipy/interpolate/tests/test_polyint.py]: Since they don't seem to be used previously, maybe simply remove it. Or add as a new separate test

**Review by ev-br — COMMENTED** [review]
Looks good, just a couple of minor suggestions. Would be nice to finish it up to keep the ball rolling.

File: scipy/interpolate/fitpack.py (hunk 2 above)
Comment (pvanmulbregt) [reply scipy/interpolate/fitpack.py]: Done.

File: scipy/interpolate/tests/test_fitpack.py (hunk 3 above)
Comment (pvanmulbregt) [reply scipy/interpolate/tests/test_fitpack.py]: Done.

File: scipy/interpolate/tests/test_interpolate.py (hunk 4 above)
Comment (pvanmulbregt) [reply scipy/interpolate/tests/test_interpolate.py]: Done.

File: scipy/interpolate/tests/test_polyint.py (hunk 1 above)
Comment (pvanmulbregt) [reply scipy/interpolate/tests/test_polyint.py]: Removed.

**Comment by ev-br** [comment]
Review comments addressed, no further activity, so let's merge it in. Thanks Paul

**Pull request merged**